### PR TITLE
Added 2 new queries related to suspicious BEC file access.

### PR DESCRIPTION
**Pull request by petebryan**
Required items, please complete
   
   Change(s):
   - Added 2 new queries looking for suspicious file access patterns related to Business Email Compromise scenarios

   Reason for Change(s):
   - New content

   Version Updated:
   - N/A

   Testing Completed:
   - Yes

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

### CODE REVIEW

**Review by oshezaf — CHANGES_REQUESTED** [review]

File: Detections/ASimFileEvent/SuspiciousAccessOfBECRelatedDocuments.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  let baseline_events = (
+  imFileEvent
+  | where TimeGenerated between (ago(14d)..ago(1d))
+    // Remove anonymous access due to noise, monitor for this seperately
+  | where isnotempty(User)
+  | where User !~ "app@sharepoint"
+  | where EventType =~ "FileAccessed"
+  | extend OriginalEvent = columnifexists("EventOriginalType","Unknown")
+  | where OriginalEvent !~ "FileSyncDownloadedFull"
+  | where EventProduct in ("SharePoint 365", "Azure File Storage", "OneDrive" , "SharePoint")
```
[the remark this answers is not in the document]
Comment (petebryan) [reply Detections/ASimFileEvent/SuspiciousAccessOfBECRelatedDocuments.yaml:33]: I've responded to this via email.

File: Detections/ASimFileEvent/SuspiciousAccessOfBECRelatedDocuments.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  | extend SPuser = case(gettype(_AuthDetails) == "array", tostring(parse_json(tostring(todynamic(_AuthDetails)[0].principals))[0].id), "Unknown")
+  | extend User = case(isnotempty(User), User, SPuser));
+  current_events
+  | summarize number_of_files_accessed = dcount(FilePath), files_accessed=make_set(FilePath, MaxSize=1000) by User, bin(TimeGenerated, 1d)
+  // Look for situations where volume of files accessed is significantly above baseline activity
+  | where number_of_files_accessed > threshold
+  | join kind=inner baseline_events on User
+  | where number_of_files_accessed > (baseline_count*deviation)
+  | join kind=leftouter current_events on User
+  | extend IpAddr = columnifexists("IpAddr", SrcIpAddr)
```
[the remark this answers is not in the document]
Comment (petebryan) [reply Detections/ASimFileEvent/SuspiciousAccessOfBECRelatedDocuments.yaml:60]: In several situations I have found IpAddr isn't present so this is needed to ensure it works across all workspaces.

File: Detections/ASimFileEvent/SuspiciousAccessOfBECRelatedDocuments.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  | project-reorder TimeGenerated, User, EventType, EventResult, EventProduct, FilePath, HttpUserAgent, IpAddr, number_of_files_accessed, baseline_count, files_accessed
```
[the remark this answers is not in the document]
Comment (oshezaf) [reply Detections/ASimFileEvent/SuspiciousAccessOfBECRelatedDocuments.yaml:61]: For 2: I meant [this](https://learn.microsoft.com/en-us/azure/sentinel/surface-custom-details-in-alerts) (and the [YAML counterpart](https://github.com/Azure/Azure-Sentinel/wiki/Query-Style-Guide#customdetails)). While projected fields appear when the user go to "events" to run the query, custom fields are now available directly in the incident experience.

**Comment by v-sabiraj** [comment]
@petebryan, there are few comments, can you please check on them, thanks.

File: Detections/ASimFileEvent/SuspiciousAccessOfBECRelatedDocuments.yaml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  let deviation = 2;
+  // Create a baseline of activity
+  let baseline_events = (
+  imFileEvent
+  | where TimeGenerated between (ago(14d)..ago(1d))
+    // Remove anonymous access due to noise, monitor for this seperately
+  | where isnotempty(User)
+  | where User !~ "app@sharepoint"
+  | where EventType =~ "FileAccessed"
+  | extend OriginalEvent = columnifexists("EventOriginalType","Unknown")
```
[the remark this answers is not in the document]
Comment (petebryan) [reply Detections/ASimFileEvent/SuspiciousAccessOfBECRelatedDocuments.yaml:31]: So this was something that was flagged by the validations here - we've asked for these edge cases to be fixed and hopefully they will be improved soon but these are needed for the pipeline to pass for now and given the minimal cost of this I will leave it in.

File: Detections/ASimFileEvent/SuspiciousAccessOfBECRelatedDocuments.yaml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+  | where OriginalEvent !~ "FileSyncDownloadedFull"
+  | where EventProduct in ("SharePoint 365", "Azure File Storage", "OneDrive" , "SharePoint")
+  | where FilePath has_any(BEC_Keywords)
+  | extend _AuthDetails = column_ifexists("AuthorizationDetails", "None")
+  | extend SPuser = case(gettype(_AuthDetails) == "array", tostring(parse_json(tostring(todynamic(_AuthDetails)[0].principals))[0].id), "Unknown")
```
[the remark this answers is not in the document]
Comment (petebryan) [reply Detections/ASimFileEvent/SuspiciousAccessOfBECRelatedDocuments.yaml:36]: 1 & 2. I've collected these elements and will look at telemetry for more detail on specific events where it's populated to understand the full range of situations. I will feed this into planning to get this updated to the parser, we can then update this query once the new parser is out.
3. Moved this down to after our User extraction.
4. In testing on large data sets parse_json was quicker to execute than parse on a consistent basis. However I have simplified this line as it was un-necessarily complex.

File: Detections/ASimFileEvent/SuspiciousAccessOfBECRelatedDocuments.yaml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+        columnName: User
+  - entityType: IP
+    fieldMappings:
+      - identifier: Address
+        columnName: IpAddr
+  - entityType: File
+    fieldMappings:
+      - identifier: Name
+        columnName: FilePath
+alertDetailsOverride:
```
[the remark this answers is not in the document]
Comment (petebryan) [reply Detections/ASimFileEvent/SuspiciousAccessOfBECRelatedDocuments.yaml:75]: Added

File: Detections/ASimFileEvent/SuspiciousAccessOfBECRelatedDocuments.yaml [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+  | where EventType =~ "FileAccessed"
+  | extend OriginalEvent = columnifexists("EventOriginalType","Unknown")
+  | where OriginalEvent !~ "FileSyncDownloadedFull"
+  | where EventProduct in ("SharePoint 365", "Azure File Storage", "OneDrive" , "SharePoint")
+  | where FilePath has_any(BEC_Keywords)
+  | extend _AuthDetails = column_ifexists("AuthorizationDetails", "None")
+  | extend SPuser = case(gettype(_AuthDetails) == "array", tostring(parse_json(tostring(todynamic(_AuthDetails)[0].principals))[0].id), "Unknown")
+  | extend User = case(isnotempty(User), User, SPuser));
+  current_events
+  | summarize number_of_files_accessed = dcount(FilePath), files_accessed=make_set(FilePath, MaxSize=1000) by User, bin(TimeGenerated, 1d)
```
[the remark this answers is not in the document]
Comment (petebryan) [reply Detections/ASimFileEvent/SuspiciousAccessOfBECRelatedDocuments.yaml:54]: Good point, added

File: Detections/ASimFileEvent/SuspiciousAccessOfBECRelatedDocuments.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (petebryan) [reply Detections/ASimFileEvent/SuspiciousAccessOfBECRelatedDocuments.yaml:61]: Understood, I've added those

File: Detections/ASimFileEvent/SuspiciousAccessOfBECRelatedDocuments.yaml [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+  | where FilePath has_any(BEC_Keywords)
+  | extend _AuthDetails = column_ifexists("AuthorizationDetails", "None")
+  | extend SPuser = case(gettype(_AuthDetails) == "array", tostring(parse_json(tostring(todynamic(_AuthDetails)[0].principals))[0].id), "Unknown")
+  | extend User = case(isnotempty(User), User, SPuser)
+    // Look for unique files accessed
+  | summarize baseline_file_count=dcount(FileName) by User, bin(TimeGenerated, 1d)
+  | summarize baseline_count = avg(baseline_file_count) by User);
+  let current_events = materialize(imFileEvent
+  | where TimeGenerated > ago(1d)
+  | where isnotempty(User)
```
[the remark this answers is not in the document]
Comment (petebryan) [reply Detections/ASimFileEvent/SuspiciousAccessOfBECRelatedDocuments.yaml:43]: Good suggestion, have re-written.

**Review by ashwin-patil — APPROVED** [review]

**Pull request merged**